### ISSUE TITLE
Max width produces a failure when trying to wrap specific lines that have long text

### ISSUE THREAD

**Issue by tony3298** [nicklockwood/SwiftFormat#632]
The following code is causing the invocation of `swiftformat` to produce the output below:
```swift
override func invalidationContext(forBoundsChange newBounds: CGRect) -> UICollectionViewLayoutInvalidationContext {
    let context = super.invalidationContext(forBoundsChange: newBounds) as! UICollectionViewFlowLayoutInvalidationContext
    context.invalidateFlowLayoutDelegateMetrics = newBounds.size != collectionView?.bounds.size
    return context
}
```
```
-- error: Failed to terminate
error: Failed to format any files.
```
Here is the config file I am using:
```
--swiftversion 5.1

--allman false
--binarygrouping 4,8
--closingparen same-line
--commas always
--conflictmarkers reject
--decimalgrouping 3,6
--elseposition same-line
--empty tuple
--exponentcase lowercase
--exponentgrouping disabled
--fractiongrouping disabled
--fragment false
--header ignore
--hexgrouping 4,8
--hexliteralcase uppercase
--ifdef indent
--importgrouping alphabetized
--indent 4
--indentcase false
--linebreaks lf
--maxwidth 100
--nospaceoperators ...,..<
--nowrapoperators =
--octalgrouping 4,8
--operatorfunc spaced
--patternlet hoist
--self remove
--selfrequired 
--semicolons inline
--specifierorder override,internal,open,public,private,private(set),fileprivate,final,static,lazy,weak
--stripunusedargs always
--tabwidth unspecified
--trailingclosures 
--trimwhitespace nonblank-lines
--wraparguments after-first
--wrapcollections before-first
--wrapparameters after-first
--xcodeindentation disabled
--disable blankLinesAtStartOfScope,numberFormatting,redundantFileprivate,sortedImports,trailingCommas
--enable isEmpty
```

I think I've narrowed down the issue and it has to do with wrapping and the `maxwidth` rule. If I up the `maxwidth` to 110, `swiftformat` succeeds with no issue. It seems like there might not be any leniency when `swiftformat` is unable to wrap a line under the `maxwidth` value, which causes it to fail. Obviously this isn't great and I would think `swiftformat` should be a little more lenient (or provide a leniency setting) and try to wrap as best as possible or leave the line as is. As it currently stands, this causes my build to fail since I'm using `swiftformat` via a build time run script. I can get around it for now, but I think wrapping in general should have more leniency and not fail outright.

**Comment by nicklockwood** [comment]
@tony3298 thanks for reporting this. SwiftFormat isn't designed to fail if it can't wrap a line, but from the "Failed to terminate" part of the error, it looks like it's got into an infinite loop. I'll try to fix asap.

**Comment by tony3298** [comment]
Awesome, happy to help. Thanks @nicklockwood !

**Comment by nicklockwood** [comment]
Fixed in 0.44.11
